src/middlewared/middlewared/plugins/pool.py

```diff
@@ -2836,7 +2836,9 @@ async def do_create(self, app, data):
         if '/' not in data['name']:
             verrors.add('pool_dataset_create.name', 'You need a full name, e.g. pool/newdataset')
```

Comment:
This branch does not define `parent_ds` that will be used below. You should probably just raise validation errors here.